Fix vtk OpenGL error on opening new instrument view

GuiMacielPereira opened the pull request:
### Description of work
Opening the new instrument view on Ubuntu (and at least some derivatives like Pop_OS) causes a crash with the following error:
```
2025-09-03 13:16:17.207 (  16.560s) [    78C4E02E9F80]     vtkOpenGLState.cxx:1787  WARN| Hardware does not support the number of textures defined.
```

Pinning vtk to 9.4.2 solves the issue.
Needed to change qt to qt-main so that mambabuild successfully solves the environment, otherwise fails.

<!-- Please provide an outline and reasoning for the work.
If there is no linked issue provide context.
-->

Closes #xxxx. <!-- One line per closed issue. -->

<!-- If issue raised by user. Do not leak email addresses.
**Report to:** [user name]
-->

### To test:

<!-- Include sufficient instructions for someone unfamiliar with the application to test.
Ok to refer back to instructions in the issue.
-->
Needs to be tested on Ubuntu:

- [x] Follow the instructions at https://developer.mantidproject.org/Packaging.html#building-a-custom-mantid-developer-environment for building a local developer environment
- [x] Run cmake with fresh argument: `cmake --preset=linux --fresh`
- [x] `cd build` and rebuild with `ninja`
- [x] Once finished, open new instrument view with any workspace

<!-- REMEMBER:
- Add labels, milestones, etc.
- Ensure the base of this PR is correct (e.g. release-next or main)
- Add release notes in separate file as per ([guidelines](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)), or justify their absence:
  *This does not require release notes* because <fill in an explanation of why>
-->

<!--  GATEKEEPER: When squashing, remove the section from HERE...  -->
---

### Reviewer

**Your comments will be used as part of the gatekeeper process.** Comment clearly on what you have checked and tested during your review. Provide an audit trail for any changes requested.

As per the [review guidelines](http://developer.mantidproject.org/ReviewingAPullRequest.html):

- Is the code of an acceptable quality? ([Code standards](http://developer.mantidproject.org/Standards/)/[GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html))
- Has a thorough functional test been performed? Do the changes handle unexpected input/situations?
- Are appropriately scoped unit and/or system tests provided?
- Do the release notes conform to the [guidelines](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html) and describe the changes appropriately?
- Has the relevant (user and developer) documentation been added/updated?

### Gatekeeper

As per the [gatekeeping guidelines](https://developer.mantidproject.org/Gatekeeping.html):

- Has a thorough first line review been conducted, including functional testing?
- At a high-level, is the code quality sufficient?
- Are the base, milestone and labels correct?
<!--  GATEKEEPER: ...To HERE  -->

GuiMacielPereira committed:
Fix vtk OpenGL error on opening instrument view

Pinning vtk to 9.4.2 solves the issue.
Needed to change qt to qt-main so that mambabuild successfully solves
the environment, otherwise fails.

GuiMacielPereira committed:
Merge branch 'main' into fix-iv-on-ubuntu

jclarkeSTFC committed:
Add vtk pin to mantidqt recipe for consistency

jclarkeSTFC committed:
Merge branch 'main' into fix-iv-on-ubuntu

jclarkeSTFC committed:
Localise use of temporary folders, remove shutil

These tests were having a problem writing to disk on Windows, so I've
moved the use of the temporary folder to a specific context where it is
needed, hopefully will resolve the problem. As a side effect I could
remove the shutil import.

jclarkeSTFC commented:
The failing tests on Windows all try to write an output file to disk, and none of them use any graphics, I don't see how a `vtk` pin would break them, I'm going to just put the pin on Linux since that's where the original problem is.

jclarkeSTFC committed:
Only pin vtk on Linux

jclarkeSTFC committed:
Skip tests on Windows

These tests write out PDFs with plots in, and for unknown reasons they
fail when running the Windows CI. Impossible to reproduce locally and
the algorithms work fine when run in Workbench.

RichardWaiteSTFC reviewed (approved):
Changes to single-crystal integration/plotting tests look good - I'm OK to skip the plotting tests  on windows  for now. Note we tested the windows package and the plotting did work.
Changes to .yaml files look OK to me but perhaps someone from core should check @thomashampson ?

thomashampson reviewed (approved):
I am happy with the recipe changes
